1) add an animation 2)correct mistaces (more about that https://githu…

AlekseyFeofilov opened the pull request:
1) add an animation 
2) correct mistaces from first review

AlekseyF8 committed:
1) add an animation 2)correct mistaces (more about that https://github.com/AlekseyFeofilov/Short-story/issues/1)

AlekseyFeofilov commented on app/src/main/res/values/styles.xml:
```diff
@@ -3,8 +3,12 @@
         <item name="android:visibility">invisible</item>
+        <item name="autoSizeMaxTextSize">55sp</item>
+        <item name="autoSizeMinTextSize">10sp</item>
+        <item name="autoSizeStepGranularity">15sp</item>
```
Здесь лучше, конечно, поставить 5 sp

LidiaIvanova commented on app/src/main/java/com/example/shortstory/JSONLoading.kt:
```diff
@@ -0,0 +1,34 @@
+    var json: String? = null
+
+    json = try {
+        val appearanceInfoByJSON = assets.open("appearanceInfo.json")
```
Лучше вынести эту строку в константу.

LidiaIvanova commented on app/src/main/java/com/example/shortstory/MainActivity.kt:
```diff
@@ -33,54 +38,66 @@ class MainActivity : AppCompatActivity() {
-            startActivity(intent)
+            button.startAnimation(AnimationUtils.loadAnimation(this, R.anim.hide))
+
+            Handler(Looper.getMainLooper()).postDelayed({
```
Такой способ уже устарел. Лучше использовать корутины. Самая простая и понятная в этой вселенной лекция по корутинам https://www.youtube.com/watch?v=aFYwPFje3uk&t=671s

LidiaIvanova commented on app/src/main/java/com/example/shortstory/MainActivity.kt:
```diff
@@ -33,54 +38,66 @@ class MainActivity : AppCompatActivity() {
-        textView.text = if (name == null) activity.getString(textViewName)
-        else activity.getString(textViewName).format(name)
+    private fun setTextView(textView: TextView, text: String) {
+        textView.text = if (name == null) text
```
Вместо такого if лучше использовать Элвис-оператор https://kotlinlang.org/docs/null-safety.html#elvis-operator

AlekseyFeofilov replied:
Тут у меня name используется внутри text.format(), поэтому я что-то не могу придумать, как с Элвис-оператором сделать это. При вызове text.format(null) либо ничего не произойдёт, если форматирование и не требовалось, либо MissingFormatArgumentException, а не NullPointerException

AlekseyF8 committed:
1) use coroutine except Handler and Looper
2) extract all string
3) fix naming and follow a bit OOP principles
4) try to use GSON
5) change json file and ActivityStructure class
